### PR TITLE
Add test groupby complex numbers

### PR DESCRIPTION
**Pull request by aram-cedarwood**
- [x] closes https://github.com/pandas-dev/pandas/issues/26475
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by aram-cinnamon** [comment]
@phofl

**Comment by aram-cinnamon** [comment]
I can change the name of the function or place it elsewhere, lmk what you think

File: pandas/tests/groupby/test_groupby.py [hunk 1]
```diff
@@ -1196,6 +1196,21 @@ def test_groupby_complex():
     tm.assert_series_equal(result, expected)
 
 
+def test_groupby_complex_2():
```
Comment (phofl) [inline pandas/tests/groupby/test_groupby.py:1199]: A name that describes the issue a little would be great

Note: We add 

# GH#issue number

as the first line of the test

File: pandas/tests/groupby/test_groupby.py [hunk 2]
```diff
@@ -1196,6 +1196,21 @@ def test_groupby_complex():
...
+    df = DataFrame(
+        [
+            {"a": 2, "b": 1 + 2j},
+            {"a": 1, "b": 1 + 1j},
+            {"a": 1, "b": 1 + 2j},
+        ]
+    )
+    assert df.groupby("b").groups == {(1 + 1j): [1], (1 + 2j): [0, 2]}
```
Comment (phofl) [inline pandas/tests/groupby/test_groupby.py:1207]: You don't have to assert the groups, asserting the final result like you do below is sufficient

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @aram-cinnamon

**Pull request merged**